src/JitInterface/src/CorInfoTypes.cs

```diff
@@ -1180,139 +1180,22 @@ public struct ILVarInfo
         public uint varNumber;
     };
 
+    [StructLayout(LayoutKind.Sequential)]
```

Comment:
I'm not sure whether you need any change of the following data structure at all.
With regard to serialize the arrays of NativeVarInfo that you hacked in the EmitDebugVar above, you could something like:
```
public static byte[] GetBytes(NativeVarInfo[] nativeVarInfos)
{    
    int unitSize = Marshal.SizeOf(typeof(NativeVarInfo));
    int len = nativeVarInfos.Length;
    int size =  unitSize * len;
    byte[] bytes= new byte[size];

    IntPtr ptr = Marshal.AllocHGlobal(unitSize);
    for(int i=0 ; i < len; i++)
    {
        Marshal.StructureToPtr(nativeVarInfos[i], ptr, true);
        Marshal.Copy(ptr, bytes, unitSize * i, unitSize);
    }
    Marshal.FreeHGlobal(ptr);

    return bytes;
}
```